### DISCUSSION TITLE
Where and what is its function?

### DISCUSSION THREAD

**Discussion by AlanJair** [pimcore/pimcore#14719] [Q&A / Support]
Hello I just entered the world of pimcore and symfony and would like to know what they work for (or a brief description) and where to find each of these:

```php
use App\Services\SegmentTrackingHelperService;
use App\Website\LinkGenerator\ProductLinkGenerator;
use App\Website\Navigation\BreadcrumbHelperService;
use Knp\Component\Pager\Pagination\SlidingPagination;
use Knp\Component\Pager\PaginatorInterface;
use Pimcore\Bundle\EcommerceFrameworkBundle\Factory;
use Pimcore\Bundle\EcommerceFrameworkBundle\FilterService\ListHelper;
use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\ProductList\DefaultMysql;
use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\ProductList\ElasticSearch\AbstractElasticSearch;
use Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\ProductList\ProductListInterface;
use Pimcore\Config;
use Pimcore\Model\DataObject\AbstractObject;
use Pimcore\Model\DataObject\Concrete;
use Pimcore\Model\DataObject\Data\UrlSlug;
use Pimcore\Model\DataObject\FilterDefinition;
use Pimcore\Translation\Translator;
use Pimcore\Twig\Extension\Templating\HeadTitle;
use Pimcore\Twig\Extension\Templating\Placeholder;
``` 

An apology in advance for my English.

**Comment by NiklasBr** [comment]
What you are looking at are [imports](https://www.php.net/manual/en/language.namespaces.importing.php) and namespaces which are regular PHP language constructs.

If you have followed the [installation instructions](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/index.html) and used [Composer](https://getcomposer.org/) you should find the code in the `vendor` directory.

See for example:
- [Documentation for Symfony](https://symfony.com/doc/5.4/index.html)
- [Documentation for KNP Component](https://github.com/KnpLabs/KnpPaginatorBundle)
